### PR TITLE
activity share

### PR DESCRIPTION
**Pull request by muhammadqamar**

### CODE REVIEW

**Review by simonebelotti75 — CHANGES_REQUESTED** [review]

File: src/components/ResourceCard/index.js [hunk 1]
```diff
@@ -144,7 +147,9 @@ const ResourceCard = (props) => {
                                       allowOutsideClick: false,
                                     });
                                   }}
-                                />
+                                >
+                                  <FontAwesomeIcon icon="clipboard" />
+                                </i>
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/index.js:152]: we are using a higher version so they generate SVG. however, I have changed tag as well

File: src/components/ResourceCard/index.js [hunk 2]
```diff
@@ -84,17 +85,19 @@ const ResourceCard = (props) => {
...
-                        {' '}
                         Edit
                       </Link>
-                      <a
+
+                      <Dropdown.Item
                         className="dropdown-item"
-                        onClick={() => {
+                        onClick={async () => {
+                          ActiveShareActivity(resource.id);
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/index.js:100]: this is for toggle active status flag... by default all resource are not sharable

File: src/components/ResourceCard/index.js [hunk 3]
```diff
@@ -84,17 +85,19 @@ const ResourceCard = (props) => {
                         <FontAwesomeIcon icon="eye" />
                         Preview
                       </Link>
+
                       <Link
                         className="dropdown-item"
                         to={`/project/${match.params.projectId}/playlist/${playlist.id}/activity/${resource.id}/edit`}
                       >
                         <FontAwesomeIcon icon="pen" />
-                        {' '}
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/index.js:92]: done

File: src/components/ResourceCard/index.js [hunk 4]
```diff
@@ -176,14 +182,13 @@ const ResourceCard = (props) => {
                         }}
                       >
                         <FontAwesomeIcon icon="times-circle" />
-                        {' '}
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/index.js:179]: done

File: src/components/ResourceCard/index.js [hunk 5]
```diff
@@ -176,14 +182,13 @@ const ResourceCard = (props) => {
...
                         Executable
-                      </a>
-                      <a className="dropdown-item" onClick={handleDelete}>
+                      </Dropdown.Item>
+
+                      <Dropdown.Item className="dropdown-item" onClick={handleDelete}>
                         <FontAwesomeIcon icon="times-circle" />
-                        {' '}
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/index.js:184]: done

File: src/components/ResourceCard/style.scss [hunk 6]
```diff
@@ -4,4 +4,24 @@
...
-}
\ No newline at end of file
+}
+
+.share-project-preview-url.project-share-check {
+  h3 {
+        text-align: center;
+        padding-bottom: 10px;
+        font-size: 16px;
+        font-weight: 400;
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/style.scss:14]: done

File: src/components/ResourceCard/style.scss [hunk 7]
```diff
@@ -4,4 +4,24 @@
...
+  }
+
+  input {
+        width: 90%;
+        border: none;
+        cursor: pointer;
+        color: #007bff;
+        text-align: center;
+        margin-bottom: 10px;
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/style.scss:23]: done

File: src/services/resource.service.js [hunk 8]
```diff
@@ -73,6 +73,16 @@ const activityH5p = (activityId) => httpService
   .then(({ data }) => data)
   .catch((err) => Promise.reject(err.response.data));
 
+const activeShare = (activityId) => httpService
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/services/resource.service.js:76]: done

File: src/routers/OpenRouter.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+OpenRoute.propTypes = {
+  component: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.func,
+  ]).isRequired,
+  path: PropTypes.string.isRequired,
+  previewType: PropTypes.string.isRequired,
+};
+
+export default OpenRoute;
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/routers/OpenRouter.js:22]: yes, we will be implementing a couple of shared routes, like Lti, etc.  kindly use this component for that

File: src/containers/Preview/PlaylistPreview/ActivityShared.js [hunk 10]
```diff
@@ -1,66 +1,86 @@
...
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import { loadh5pShareResource } from 'store/actions/resource';
 import { withRouter } from 'react-router-dom';
 
-import { loadPlaylistAction } from 'store/actions/playlist';
-
 import './style.scss';
+import gifloader from 'assets/images/dotsloader.gif';
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/containers/Preview/PlaylistPreview/ActivityShared.js:7]: done

File: src/containers/Preview/PlaylistPreview/ActivityShared.js [hunk 11]
```diff
@@ -1,66 +1,86 @@
-import React, { Suspense } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import { loadh5pShareResource } from 'store/actions/resource';
 import { withRouter } from 'react-router-dom';
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/containers/Preview/PlaylistPreview/ActivityShared.js:4]: done

File: src/store/actions/resource.js [hunk 12]
```diff
@@ -505,7 +505,7 @@ export const onSubmitDescribeActivityAction = (metadata, activityId = null) => (
       },
     });
   } catch (e) {
-    console.log(e);
+    return e;
```
Comment (simonebelotti75) [inline src/store/actions/resource.js:508]: Same here

File: src/store/actions/resource.js [hunk 13]
```diff
@@ -543,3 +543,13 @@ export const uploadResourceThumbnailAction = (formData) => async (dispatch) => {
     // throw new Error(e);
   }
 };
+
+export const activeShareActivity = async (actvityId) => {
```
Comment (simonebelotti75) [inline src/store/actions/resource.js:547]: I think function name should be shareActivity
Also it seems like result is redundant

File: src/store/actions/resource.js [hunk 14]
```diff
@@ -543,3 +543,13 @@ export const uploadResourceThumbnailAction = (formData) => async (dispatch) => {
...
+  const result = await resourceService.shareActivity(actvityId);
+  return result;
+};
+
+export const loadh5pShareResource = async (actvityId) => {
+  const result = await resourceService.loadH5pShared(actvityId);
+  return result;
```
Comment (simonebelotti75) [inline src/store/actions/resource.js:554]: result is redundant here

**Review by simonebelotti75 — CHANGES_REQUESTED** [review]

File: src/store/actions/resource.js (hunk 14 above)
Comment (muhammadqamar) [reply src/store/actions/resource.js:554]: no we need this for H5P rendering

File: src/store/actions/resource.js (hunk 13 above)
Comment (muhammadqamar) [reply src/store/actions/resource.js:547]: done

File: src/store/actions/resource.js (hunk 12 above)
Comment (muhammadqamar) [reply src/store/actions/resource.js:508]: done

File: src/store/actions/resource.js [hunk 15]
```diff
@@ -490,7 +490,7 @@ export const onChangeActivityAction = (activity) => (dispatch) => {
       payload: { activity },
     });
   } catch (e) {
-    console.log(e);
+    return e;
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/store/actions/resource.js:493]: done

File: src/components/ResourceCard/index.js [hunk 16]
```diff
@@ -119,7 +122,7 @@ const ResourceCard = (props) => {
                                   />
                                 </a>
 
-                                <i
+                                <span
                                   title="copy to clipboard"
                                   className="fa fa-clipboard"
                                   aria-hidden="true"
```
[the remark this answers is not in the document]
Comment (muhammadqamar) [reply src/components/ResourceCard/index.js:128]: classname should not be there and it is removed.
we are doing some other operation as well.. e.g copy to clipboard , focusing a ref etc

**Review by simonebelotti75 — APPROVED** [review]

**Pull request merged**